Plugins/Gerrit/FormGerritBase.cs

```diff
@@ -66,7 +66,15 @@ private Version GetGerritVersion()
         {
             RestRequest request = new RestRequest("/config/server/version");
             IRestResponse response = client.Execute(request);
-            return Version.Parse(response.Content.Replace(")]}'", "").Replace("\n", "").Replace("\"", "").Split('-')[0]);
+            if (response.IsSuccessful)
+            {
+                string versionText = response.Content.Replace(")]}'", "").Replace("\n", "").Replace("\"", "").Split('-')[0];
+                return Version.Parse(versionText);
```

Comment:
Unless `versionText` conforms to a string parseable to `Version` this will fail again.

This is where a regex may help, because you can enforce the format (though not perfectly): https://docs.microsoft.com/en-us/dotnet/api/system.version.-ctor?view=netframework-4.8

```^(\d*\.\d*(\.\d*){0,2})$```
![image](https://user-images.githubusercontent.com/4403806/71456427-7ad91180-27ed-11ea-892a-c93b62b9ba40.png)
Here I am obviously testing for a string from a beginning and to the end of the line, given I have no idea `/config/server/version` return payload is... 

However `Version.Parse(string)` call must still be wrapped in `try..catch` https://docs.microsoft.com/en-us/dotnet/api/system.version.parse?view=netframework-4.8


So the method should look something like the following:

```cs
            RestRequest request = new RestRequest("/config/server/version");
            IRestResponse response = client.Execute(request);
            if (!response.IsSuccessful)
			{
				return GerritUtil.GerritWithPrivateReviewSupport;
			}

			try
			{
				var m = Regex.Match(..., ...);
				if (m.Success)
				{
					return Version.Parse(m....);
				}
			}
			catch
			{
				// no-op
			}

			return GerritUtil.GerritWithPrivateReviewSupport;
```

I would also encourage you to write tests to test edge-cases.

Reply:
changed to regex matching

Reply:
@RussKie on testing, the only feasible way would be to extract the version retrieval part to a separate class and mock the RestClient. (
`class GerritServer { 
public GerritServer(IRestClient client, GerritSettings settings)
{...}
public Version GetVersion();
{...}
}`
Then the GerritSettings class would also need some changes  as it not test friendly (cannot be stubbed and requires file parsing under a IGitModule context)

So, unit tests would be indeed nice, but right now seem a little to hard to write. So it will take some time....